app/waf/src/utils.ts

```diff
@@ -0,0 +1,41 @@
+import {factory} from './log4ts';
+
+const utilsLogger = factory.getLogger('utils.logger');
+
+export async function checkAndWait(
+  checkFunc: Function,
+  tryTimes: number,
+  funcArgs: (object | string | number | boolean | undefined)[] = [],
+  intervalInMSecs: number = 1000,
+): Promise<void> {
+  let funcName = checkFunc.name ? checkFunc.name : 'anonymous';
+
+  utilsLogger.debug(
+    `Check and wait '${funcName}' to return, countdown: ${tryTimes}.`,
+  );
+  if (tryTimes <= 0) {
+    utilsLogger.debug(`'${funcName}' timeout.`);
+    return Promise.reject();
+  }
+
+  let hdlr = async (b: boolean): Promise<boolean> => {
+    if (typeof b === 'boolean' && b) {
+      utilsLogger.debug(`'${funcName}' returns ${b}, countdown: ${tryTimes}`);
+      return Promise.resolve(true);
+    } else throw new Error();
+  };
+
+  try {
+    await checkFunc(...funcArgs).then(hdlr, hdlr);
```

Comment:
I do not know is there any boolean parameters will be passed to this async function?

Reply:
Didn't get your point...
The `checkFunc` is supposed to return `Promise<boolean>`, which is resolved/rejected to `hdlr`, then return nothing in `checkAndWait` context. 
And the `checkAndWait` function returns `Promise<void>`(resolve when `checkFunc` condition is fulfilled; reject when `timeout`.)